Sources/StreamChat/Database/DTOs/PollDTO.swift

```diff
@@ -0,0 +1,180 @@
+        request.predicate = NSPredicate(format: "id == %@", pollId)
+        return request
+    }
+}
+
+extension PollDTO {
+    func asModel() throws -> Poll {
+        var extraData: [String: RawJSON] = [:]
+        if let custom,
+           !custom.isEmpty,
+           let decoded = try? JSONDecoder.default.decode([String: RawJSON].self, from: custom) {
+            extraData = decoded
+        }
+        
+        let optionsArray = (options.array as? [PollOptionDTO]) ?? []
+        
+        return try Poll(
+            allowAnswers: allowAnswers,
+            allowUserSuggestedOptions: allowUserSuggestedOptions,
+            answersCount: answersCount,
+            createdAt: createdAt.bridgeDate,
+            pollDescription: pollDescription,
+            enforceUniqueVote: enforceUniqueVote,
+            id: id,
+            name: name,
+            updatedAt: updatedAt?.bridgeDate,
+            voteCount: voteCount,
+            extraData: extraData,
+            voteCountsByOption: voteCountsByOption,
+            isClosed: isClosed,
+            maxVotesAllowed: maxVotesAllowed?.intValue,
+            votingVisibility: votingVisibility(from: votingVisibility),
+            createdBy: createdBy?.asModel(),
+            latestAnswers: latestAnswers.map { try $0.asModel() },
+            options: optionsArray.map { try $0.asModel() },
+            latestVotesByOption: latestVotesByOption.map { try $0.asModel() }
+        )
+    }
+    
+    private func votingVisibility(from string: String?) -> VotingVisibility? {
+        guard let string else { return nil }
+        return VotingVisibility(rawValue: string)
+    }
+}
+
+extension NSManagedObjectContext {
+    @discardableResult
+    func savePoll(payload: PollPayload, cache: PreWarmedCache?) throws -> PollDTO {
+        let pollDto = PollDTO.loadOrCreate(pollId: payload.id, context: self, cache: cache)
+        
+        pollDto.allowAnswers = payload.allowAnswers
+        pollDto.allowUserSuggestedOptions = payload.allowUserSuggestedOptions
+        pollDto.answersCount = payload.answersCount
+        pollDto.createdAt = payload.createdAt.bridgeDate
+        pollDto.pollDescription = payload.description
+        pollDto.enforceUniqueVote = payload.enforceUniqueVote
+        pollDto.name = payload.name
+        pollDto.updatedAt = payload.updatedAt.bridgeDate
+        pollDto.voteCount = payload.voteCount
+        pollDto.custom = try JSONEncoder.default.encode(payload.custom)
+        pollDto.voteCountsByOption = payload.voteCountsByOption
+        pollDto.isClosed = payload.isClosed ?? false
+        if let maxVotesAllowed = payload.maxVotesAllowed {
+            pollDto.maxVotesAllowed = NSNumber(value: maxVotesAllowed)
+        }
+        pollDto.votingVisibility = payload.votingVisibility
+        
+        pollDto.createdBy = UserDTO.loadOrCreate(id: payload.createdById, context: self, cache: cache)
```

Comment:
Seems like we missing a line here which stores the user. `payload.createdBy` is `UserPayload` and we are not saving it here so this can return nil.